test/android/core/javatests/src/org/xwalk/core/xwview/test/SettingsTest.java

```diff
@@ -17,10 +19,80 @@
     private static final String USER_AGENT =
             "Chrome/44.0.2403.81 Crosswalk/15.44.376.0 Mobile Safari/537.36";
     private static final String LANGUAGE = "jp";
+    private TestWebServer mWebServer;
 
     @Override
     public void setUp() throws Exception {
         super.setUp();
+        mWebServer = TestWebServer.start();
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        mWebServer.shutdown();
+        super.tearDown();
+    }
+
+    @MediumTest
+    @Feature({"Settings"})
+    public void testCacheMode() throws Throwable {
+        final String htmlPath = "/testCacheMode.html";
+        final String url = mWebServer.setResponse(htmlPath, "response", null);
+        final String htmlNotInCachePath = "/testCacheMode-not-in-cache.html";
+        final String urlNotInCache = mWebServer.setResponse(htmlNotInCachePath, "", null);
+
+        clearCacheOnUiThread(true);
+        assertEquals(XWalkSettings.LOAD_DEFAULT, getCacheMode());
+
+        setCacheMode(XWalkSettings.LOAD_CACHE_ELSE_NETWORK);
+        loadUrlSync(url);
+        assertEquals(1, mWebServer.getRequestCount(htmlPath));
+        loadUrlSync(url);
+        assertEquals(1, mWebServer.getRequestCount(htmlPath));
+
+        setCacheMode(XWalkSettings.LOAD_NO_CACHE);
+        loadUrlSync(url);
+        assertEquals(2, mWebServer.getRequestCount(htmlPath));
+        loadUrlSync(url);
+        assertEquals(3, mWebServer.getRequestCount(htmlPath));
+
+        setCacheMode(XWalkSettings.LOAD_CACHE_ONLY);
+        loadUrlSync(url);
+        assertEquals(3, mWebServer.getRequestCount(htmlPath));
+        loadUrlSync(url);
```

Comment:
No, it's necessary.
LOAD_CACHE_ELSE_NETWORK means "use cached resources when they are available, even if they have expired. Otherwise load resources from the network."
So after the 1st loadUrlSync(page cached), we should verify the 2nd loadUrlSync is from cache instead of network.